build_runner/lib/src/generate/build_impl.dart

```diff
@@ -256,7 +256,9 @@ class BuildImpl {
 
       // Yield the outputs.
       for (var output in wrappedWriter.assetsWritten) {
-        (_assetGraph.get(output) as GeneratedAssetNode).wasOutput = true;
+        var node = _assetGraph.get(output) as GeneratedAssetNode;
+        node.wasOutput = true;
+        node.globs = wrappedReader.globsRan.toSet();
```

Comment:
actually reading over this I will need to do this even if it _wasn't_ output so I will follow up with another test and fix